animation_nodes/algorithms/mesh_generation/find_shortest_path.pyx

```diff
@@ -0,0 +1,126 @@
+from ... data_structures import GPStroke
+from ... data_structures cimport (
+    Mesh,
+    LongList,
+    ColorList,
+    FloatList,
+    PolySpline,
+    DoubleList,
+    Vector3DList,
+)
+
+from . line import getLinesMesh
+from ... math cimport distanceVec3
+
+# Dijkstra's algorithm (https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm) but is implemented
+# such a way to handle multi-sources and mesh with multiple islands.
+def getShortestPath(Mesh mesh, LongList sources, LongList targets, str pathType, str mode):
+    cdef Vector3DList vertices = mesh.vertices
+    cdef Py_ssize_t vertexCount = vertices.length
+    cdef float maxWeight = 1000000
+    cdef DoubleList weights = DoubleList.fromValue(maxWeight, length = vertexCount)
+    cdef LongList previousVertices = LongList.fromValue(-1, length = vertexCount)
+    cdef LongList visitedVertices = LongList.fromValue(0, length = vertexCount)
+    cdef Py_ssize_t i
+
+    for i in range (sources.length):
+        weights.data[sources.data[i]] = 0.0
+
+    cdef LongList linkedVertices
+    cdef float weight, minWeight
+    cdef Py_ssize_t k, l, linkedIndex, currentIndex
+
+    for i in range(vertexCount):
+
+        minWeight = maxWeight
+        for k in range(vertexCount):
+            weight = weights.data[k]
+            if weight < minWeight and visitedVertices.data[k] == 0:
+                minWeight = weight
+
+        if minWeight == maxWeight:
+            break
+
+        for currentIndex in range(vertexCount):
+            if abs(minWeight - weights.data[currentIndex]) < 1.0e-6:
+                visitedVertices.data[currentIndex] = 1
+
+                linkedVertices = mesh.getVertexLinkedVertices(currentIndex)[0]
+
+                for l in range(linkedVertices.length):
```

Comment:
Shouldn't we filter the neighbors based on the visited array instead?

Reply:
Omar, We have to update the weight/distance of each neighbor vertex and update their previous vertex index. The current vertex can have visited and non-visited vertices.

Reply:
Ok. I will look at this in more details once I get the time.

Reply:
Okay.